src/System.ComponentModel.TypeConverter/src/System/ComponentModel/InterlockedBitVector32.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.ComponentModel
+{
+    /// <summary>
+    /// Provides a subset of the <see cref="BitVector32"/> surface area, using volatile operations for reads and interlocked operations for writes. 
+    /// </summary>
+    internal struct InterlockedBitVector32
+    {
+        private int _data;
+
+        public InterlockedBitVector32(int data)
+        {
+            _data = data;
+        }
+
+        public bool this[int bit]
+        {
+            get { return (Volatile.Read(ref _data) & bit) == bit; }
+            set
+            {
+                while (true)
+                {
+                    int oldValue = _data;
+                    int newValue = value ? oldValue | bit : oldValue &= ~bit;
+                    if (Interlocked.CompareExchange(ref _data, newValue, oldValue) == oldValue)
+                    {
+                        break;
+                    }
+                }
+            }
+        }
+
+        /// <summary>Sets or unsets the specified bit, without using interlocked operations.</summary>
+        public void DangerousSet(int bit, bool value)
+        {
+            _data = value ? _data | bit : _data & ~bit;
+        }
+
+        public int Data => (int)_data;
```

Comment:
Just wondering is this cast to int necessary? It seems that _data is already an int